src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public partial class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+        internal static readonly MediaTypeHeaderValue DefaultMediaType = MediaTypeHeaderValue.Parse(string.Format("{0} {1}", JsonMediaType, Encoding.UTF8.WebName));
+
+        private readonly JsonSerializerOptions? _jsonSerializerOptions;
+        public Type ObjectType { get; }
+        public object? Value { get; }
+
+        private JsonContent(object? value, Type inputType, MediaTypeHeaderValue mediaType, JsonSerializerOptions? options)
+        {
+            if (mediaType == null)
+            {
+                throw new ArgumentNullException(nameof(mediaType));
+            }
+
+            if (inputType == null)
+            {
+                throw new ArgumentNullException(nameof(inputType));
+            }
+
+            Value = value;
+            ObjectType = inputType;
+            Headers.ContentType = mediaType;
+            _jsonSerializerOptions = options;
+        }
+
+        public static JsonContent Create<T>(T value, MediaTypeHeaderValue mediaType, JsonSerializerOptions? options = null)
+        {
+            return Create(value, typeof(T), mediaType, options);
```

Comment:
In the case of T == typeof(object) and value != null, shouldn't we be using value.GetType() as the type?  Or is that handled somewhere else? @bartonjs, is that correct?

Reply:
That is fair since that way we honor always using the runtime type; however that is contrary to how JsonSerializer works, for generics, it uses the declared type but since we are boxing anyway (and we always use the non-generic API), I think if fine for us to differ from it.

Reply:
> however that is contrary to how JsonSerializer works, for generics, it uses the declared type.

If that's true, then do we need to revisit the overloads decision from the API review meeting?  We decided it was ok to go with just the generic overloads for the various methods with the idea that if you passed an arbitrary object it would use the actual type.
cc: @terrajobst, @bartonjs, @GrabYourPitchforks

Reply:
David and I spoke offline; the serializer does in fact use GetType when T == object, so it's as we discussed in API review.  In that case, we can just use typeof(T) always.